agent/tcs/client/client.go

```diff
@@ -237,7 +237,21 @@ func (cs *clientServer) metricsToPublishMetricRequests(includeServiceConnectStat
 	for i, taskMetric := range taskMetrics {
 		requestMetadata = copyMetricsMetadata(metadata, false)
 
-		// TODO [SC]: Check if SC is enabled for the task
+		// Check if taskMetric without service connect metrics exceed the message size
+		tempTaskMetric := *taskMetric
+		tempTaskMetric.ServiceConnectMetricsWrapper = tempTaskMetric.ServiceConnectMetricsWrapper[:0]
```

Comment:
Should we only assign this if `includeServiceConnectStats` is true?

Reply:
Synced offline - we are setting it to an empty slice here to make sure the size calculation is pre-SC.